howso/details.amlg

```diff
@@ -401,61 +461,119 @@
 			(if (>= model_size 2)
-				(call !ComputeFeatureContributions)
+				(call !ComputeFeatureContributions (assoc robust (true)))
 				;can't compute contributions if model is too small
-				(accum (assoc output (assoc "feature_contributions" (zip context_features 0)) ))
+				(accum (assoc output (assoc "feature_contributions_robust" (zip context_features 0)) ))
 			)
 		)
 
-		(if (get details "case_feature_contributions")
+		(if (get details "feature_contributions_full")
 			(if (>= model_size 2)
-				(call !ComputeCaseFeatureContributions)
+				(call !ComputeFeatureContributions (assoc robust (false)))
 				;can't compute contributions if model is too small
-				(accum (assoc output (assoc "case_feature_contributions" (zip context_features 0)) ))
+				(accum (assoc output (assoc "feature_contributions_full" (zip context_features 0)) ))
 			)
 		)
 
-		(if (get details "case_contributions")
-			(if robust_influences
-				(call !ComputeCaseContributionsRobust)
+		(if (get details "case_feature_contributions_robust")
+			(if (>= model_size 2)
+				(call !ComputeCaseFeatureContributions (assoc robust (true)))
+				;can't compute contributions if model is too small
+				(accum (assoc output (assoc "case_feature_contributions_robust" (zip context_features 0)) ))
+			)
+		)
 
-				(call !ComputeCaseContributions)
+		(if (get details "case_feature_contributions_full")
+			(if (>= model_size 2)
+				(call !ComputeCaseFeatureContributions (assoc robust (false)))
+				;can't compute contributions if model is too small
+				(accum (assoc output (assoc "case_feature_contributions_full" (zip context_features 0)) ))
 			)
 		)
 
-		(if (get details "case_mda")
-			(call !ComputeCaseMDA)
+		(if (get details "case_contributions_robust")
+			(call !ComputeCaseContributionsRobust)
 		)
 
-		;If the confusion matrices are selected, then all of the stats are returned.
-		(if (get details "confusion_matrices")
-			(accum (assoc details (assoc "prediction_stats" (true))))
+		(if (get details "case_contributions_full")
+			(call !ComputeCaseContributionsFull)
+		)
+		(if (get details "case_mda_robust" )
+			(call !ComputeCaseMDA (assoc robust (true)))
 		)
 
-		;append feature errors and deviations
-		(if (or (get details "prediction_stats") (get details "feature_residuals"))
-			(declare 
-				(assoc 
-					compute_all_statistics (false)
+		(if (get details "case_mda_full" )
+			(call !ComputeCaseMDA (assoc robust (false)))
+		)
+
+		;update the 'selected_prediction_stats' if none is provided, or 'all is provided. If prediction_stats is not selected, empties the list
+		(if (or (get details "prediction_stats") (get details "prediction_stats_robust"))
+			(seq
+				(if (= 0 (size selected_prediction_stats))
+					(assign (assoc
+						selected_prediction_stats
+							(filter (lambda (!= "confusion_matrix" "all" (current_value))) !supportedPredictionStats)
+					))
+					;elif
+					(if (contains_value selected_prediction_stats "all")
```

Comment:
I may be misunderstanding the intention here, but two things. 

If the intention here is really an "elif", you can just list an additional condition/branch like
```
(if (= 0 (size selected_prediction_stats))
     (assign (assoc blah blah ))

     ;elif
     (contains_value selected_prediction_stats "all")
     (assign (assoc blah blah))
)
```

However, I think the even better option here is to be a bit more functional. All these branches assign values into selected prediction stats, so put the `(if)` branching logic *within* the assignment. Like so:
```
(assign (assoc
     selected_prediction_stats
          (if (= 0 (size selected_prediction_stats))
               (filter (lambda (....

               (contains_value selected_prediction_stats "all")
               (filter (lambda (!= "all" (current_value) .....
          )
))
```

Feel free to reach out if I'm not making any sense

Reply:
also please forgive my spacing/styling, I can't use tab in this editor...